### ISSUE TITLE
Internal Server Error due to violates not-null constraint

### ISSUE THREAD

**Issue by br95187** [WeblateOrg/weblate#5926]
**Describe the issue**

The error "Internal Server Error" is shown at home screen.

**I already tried**

The error message already is already shown at home screen, you don't need to do anything else.

- [x] I've read and searched [the docs](https://docs.weblate.org/) and did not find the answer there.
      If you didn’t try already, try to search there what you wrote above.

**To Reproduce the issue**

1. Enter home screen.

**Expected behavior**

The website displays a message of "Internal Server Error" and the app's log prints the Exception traceback below.

**Exception traceback**

```
uwsgi stderr | [2021-04-26 17:50:20,368: ERROR/392] Internal Server Error: /
uwsgi stderr | Traceback (most recent call last):
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/backends/utils.py", line 84, in _execute
uwsgi stderr |     return self.cursor.execute(sql, params)
uwsgi stderr | psycopg2.errors.NotNullViolation: null value in column "secondary" of relation "metrics_metric" violates not-null constraint
uwsgi stderr | DETAIL:  Failing row contains (636, 2021-04-26, 0, 0, projects, 1, null).
uwsgi stderr |
uwsgi stderr |
uwsgi stderr | The above exception was the direct cause of the following exception:
uwsgi stderr |
uwsgi stderr | Traceback (most recent call last):
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/exception.py", line 47, in inner
uwsgi stderr |     response = get_response(request)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py", line 181, in _get_response
uwsgi stderr |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
uwsgi stderr |     response = view_func(request, *args, **kwargs)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/dashboard.py", line 205, in home
uwsgi stderr |     return dashboard_anonymous(request)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/dashboard.py", line 328, in dashboard_anonymous
uwsgi stderr |     Metric.SCOPE_GLOBAL, 0, name="projects"
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/manager.py", line 85, in manager_method
uwsgi stderr |     return getattr(self.get_queryset(), name)(*args, **kwargs)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/weblate/metrics/models.py", line 46, in get_current
uwsgi stderr |     collect_global()
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/weblate/metrics/tasks.py", line 74, in collect_global
uwsgi stderr |     for name, value in data.items()
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/manager.py", line 85, in manager_method
uwsgi stderr |     return getattr(self.get_queryset(), name)(*args, **kwargs)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/query.py", line 507, in bulk_create
uwsgi stderr |     objs_without_pk, fields, batch_size, ignore_conflicts=ignore_conflicts,
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/query.py", line 1274, in _batched_insert
uwsgi stderr |     ignore_conflicts=ignore_conflicts,
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/query.py", line 1254, in _insert
uwsgi stderr |     return query.get_compiler(using=using).execute_sql(returning_fields)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/models/sql/compiler.py", line 1397, in execute_sql
uwsgi stderr |     cursor.execute(sql, params)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/backends/utils.py", line 66, in execute
uwsgi stderr |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
uwsgi stderr |     return executor(sql, params, many, context)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/backends/utils.py", line 84, in _execute
uwsgi stderr |     return self.cursor.execute(sql, params)
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/utils.py", line 90, in __exit__
uwsgi stderr |     raise dj_exc_value.with_traceback(traceback) from exc_value
uwsgi stderr |   File "/usr/local/lib/python3.7/dist-packages/django/db/backends/utils.py", line 84, in _execute
uwsgi stderr |     return self.cursor.execute(sql, params)
uwsgi stderr | django.db.utils.IntegrityError: null value in column "secondary" of relation "metrics_metric" violates not-null constraint
uwsgi stderr | DETAIL:  Failing row contains (636, 2021-04-26, 0, 0, projects, 1, null).
```

**Server configuration and status**

Weblate installation: Docker (Kubernetes > AWS EKS)

weblate@weblate-api-7978fcf4ff-9h8nm:/$ weblate list_versions
 * Weblate: 4.5.1
 * Django: 3.1.7
 * siphashc: 2.1
 * translate-toolkit: 3.3.3
 * lxml: 4.6.2
 * Pillow: 8.1.1
 * bleach: 3.3.0
 * python-dateutil: 2.8.1
 * social-auth-core: 4.1.0
 * social-auth-app-django: 4.0.0
 * django-crispy-forms: 1.11.1
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.12.2
 * django-filter: 2.4.0
 * django-appconf: 1.0.4
 * user-agents: 2.2.0
 * filelock: 3.0.12
 * setuptools: 40.8.0
 * jellyfish: 0.8.2
 * openpyxl: 3.0.6
 * celery: 5.0.5
 * kombu: 5.0.2
 * translation-finder: 2.9
 * weblate-language-data: 2021.3
 * html2text: 2020.1.16
 * pycairo: 1.16.2
 * pygobject: 3.30.4
 * diff-match-patch: 20200713
 * requests: 2.25.1
 * django-redis: 4.12.1
 * hiredis: 1.1.0
 * sentry_sdk: 1.0.0
 * Cython: 0.29.22
 * misaka: 2.1.1
 * GitPython: 3.1.14
 * borgbackup: 1.1.15
 * pyparsing: 2.4.7
 * pyahocorasick: 1.4.1
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2: 2.8.6
 * psycopg2-binary: 2.8.6
 * phply: 1.2.5
 * chardet: 4.0.0
 * ruamel.yaml: 0.16.13
 * tesserocr: 2.5.1
 * akismet: 1.1
 * boto3: 1.17.21
 * zeep: 4.0.0
 * aeidon: 1.9
 * iniparse: 0.5
 * mysqlclient: 2.0.3
 * Mercurial: 5.7
 * git-svn: 2.20.1
 * git-review: 2.0.0
 * Redis server: 5.0.5
 * PostgreSQL server: 13.1
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: xx.xx.xx.xx
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://xxx.xxx.xxx.com:6379/1, redis://xxx.xxx.xxxx.com:6379/1, regular
 * Platform: Linux 4.14.219-164.354.amzn2.x86_64 (x86_64)
 
**Weblate deploy checks**

weblate@weblate-api-7978fcf4ff-9h8nm:/$ weblate check --deploy
SystemCheckError: System check identified some issues:

CRITICALS:
?: (weblate.E027) The path /app/data/settings-override.py is owned by a different user, check your DATA_DIR settings.
	HINT: https://docs.weblate.org/en/weblate-4.5.1/admin/install.html#file-permissions

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
	HINT: https://docs.weblate.org/en/weblate-4.5.1/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
	HINT: https://docs.weblate.org/en/weblate-4.5.1/admin/backup.html
?: (weblate.I031) New Weblate version is available, please upgrade to 4.6.
	HINT: https://docs.weblate.org/en/weblate-4.5.1/admin/upgrade.html

System check identified 4 issues (1 silenced).

**Workaround Solution**

Table before:
![image](https://user-images.githubusercontent.com/75024600/116132409-eb26ca00-a6a3-11eb-9447-b5721d35253d.png)

To workaround this (and reestabilish production), I did execute this command in DB:
**ALTER TABLE weblate.metrics_metric ALTER COLUMN secondary SET DEFAULT 0;**

After that, the problem has been solved (palliative).

Table after:
![image](https://user-images.githubusercontent.com/75024600/116132496-01cd2100-a6a4-11eb-9de1-72bbfde002dd.png)

**Comment by nijel** [comment]
Seems like you are running Weblate 4.5.1 with database upgraded to 4.6...

**Comment by br95187** [comment]
You were totally right.

I probably made a mistake on Apr/14/21 and ran version 4.5.3 by mistake, which applied some migration and broke my environment.

I have now tried to run version 4.5.3-3 and 4.6-1, without success.
The login screen appears without formatting and after a long time two files (one .js and one .css) abort with an ERR_INCOMPLETE_CHUNKED_ENCODING error. But no errors in the application's log.

Tried different browsers (chrome, firefox, safari), tried flush redis, restart weblate pod.... notting helped.

I restored a DB backup of the day Apr/13/21 and it is running in version 4.5.1-1 without problems.

Thank you very much for your time and I'm sorry for the inconvenience.
